Improve tabview docs (#9)
tukaan/tukaan.github.io: src/routes/docs/layouts/tabview.md

![Image of a Tukaan tabview](/resources/docs/tabview.png)

> The `TabView` widget allows you to have multiple tabs where you can place widgets like any other `Container`. An example of how to set up a TabView widget is as follows:

Reviewer: ```suggestion
The `TabView` widget allows you to have multiple tabs where you can place widgets like any other container widget. An example of how to set up a TabView widget is as follows:
```

> The `TabView` widget allows users to have multiple tabs (which are ontainers) in which they can place widgets. An example of how to set up a TabView widget is as follows:

Reviewer: I'd like to use "you" to address the reader.
There's a typo, too.

button_2.grid(margin=20)

app.run()
```

## `Tab` Options

> The `TabView` widget allows users to create `Tab`'s as shown above that act as `Container`'s. The unique methods available to `Tab`'s and a description of each method are as follows:

Reviewer: ```suggestion
The `TabView` widget allows users to create `Tab`'s as shown above that act as container's. The unique methods available to `Tab`'s and a description of each method are as follows:
```

It's technically a container, but not literally

| Method | Description|
|-|-|
| `.select()` | Sets the `Tab` showing in the `TabView` to the `Tab` this is called on |
| `.move(new_index: int)` | Moves the `Tab` to the specified index in the ordering (left to right) of the `TabView` |
| `.hide()` | Hides the `Tab` visually but is still considered to be in the `TabView` |
| `.unhide()` | Shows the `Tab` again where it was before |

> | `.remove()` | Removes the `Tab` from the list internally |

Reviewer: ```suggestion
| `.remove()` | Removes the `Tab` from the `TabView`, and can only be shown again by calling `.append()` |
```

| `.append()` | Adds the `Tab` to the `TabView` at the end of the line |

Additionally the `Tab` also has the following unique attributes:
| Atrribute | Type | Description |
|-|-|-|
| `enabled` | `bool` | Changes the `Tab` to either be enabled or disabled |
| `title` | `str` | Changes the displayed name of the `Tab` |
| `icon` | `Icon` or `Image.Image` | Displays the given icon (left side by default) |
| `image_pos` | `str` | Sets the position of the icon in the specified direction |
| `margin` | `int` or `tuple[int, ...]` | Sets the margins of the `Tab` |

## `TabView` Options

The `TabView` also has a couple useful methods. They go as follows:

| Method| Description|
|-|-|
| `.on_tab_change(func: Callable[[Tab | None], None]) -> Callable[[], None]` | Binds a function that is called any time tab is changed |

> | `.enable_keyboard_traversal()` | Enables traversal o

Reviewer: ```suggestion
| `.enable_keyboard_traversal()` | Enables navigating the tabs with the keyboard
```

app.run()

```

> The first two blocks are fairly typical for a `Tukaan` program with the necessary import statement and app/window creation. The next block is where the `TabView` is clearly made and is also fairly standard. The next block, however, is slightly unorthodox. In this block we do the following: `tab_x = tabview.Tab("Tab x")`. Digging into the `TabView` source code we find [`self.Tab = Tab`](https://github.com/tukaan/tukaan/blob/f39bb197b1bbbb05a03e498114cc1c4dcff8b800/tukaan/widgets/tabview.py#L152-L155). If we go up in the file we then find class `Tab` which is really just a wrapper around the `Frame` class with extra methods for tabs. Because of how `Tab` is implemented, however, it creates a very clean API call that doesn't require much extra thought to understand. Moving on to the next two blocks we see that these tabs are interacted with exactly as one would work with a `Frame` which of course makes sense knowing that they are indeed a Frame on the inside. The final block is also fairly generic with the call of the `App.run()` method.

Reviewer: - I don't think it's necessary t explain every part of the code. At this point the reader is expected to know the parts where you use "fairly standard/generic", and I don't want the docs to re-explain these basic patterns of Tukaan everywhere.

- Please don't explain what the source code does, it's irrelevant. All the user should care about here is how they does things with the public API, the rest is implementation detail.

I think what this part should be about is the you can create a new tab by creating a `.Tab()`, and it works like an other container. The user can set parameters of the tab, like the title and icon, and can call different methods on it, like `.move()`, `.hide()`, or `.remove()`

Author: Got it 👍
